src/lib.rs

```diff
@@ -616,6 +616,21 @@ impl UPClientMqtt {
                 )
             })?;
 
+        // Add ttl to properties as message expiry time
+        if let Some(message_expiry_interval) = attributes.ttl {
```

Comment:
FMPOV this will duplicate the TTL in the MQTT message because the existing code using reflection on UAttributes will also add a corresponding User Property. IMHO we should replace the existing (reflection based) code with code that explicitly adds each UAttributes field to the MQTT message header.

Reply:
Ok yeah thats true! I tried to replace the reflection stuff with explicit handling of each attribute as per your updated  uattribute -> mqtt property mapping. Now there are more properties in the message than before tho (see the non passing tests) so Im wondering if the reflexive code actually didnt add all attributes correctly or if Im mistakenly adding an attribute or two that actually should not be added.